src/int/mul.rs

```diff
@@ -0,0 +1,181 @@
+//! [`Int`] multiplication operations.
+
+use core::ops::{Mul, MulAssign};
+
+use subtle::{Choice, ConstantTimeEq, CtOption};
+
+use crate::{Checked, CheckedMul, Int, Uint, Zero};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// Compute "wide" multiplication as a 3-tuple `(lo, hi, sgn)`.
+    /// The `(lo, hi)` components contain the _magnitude of the product_, with sizes
+    /// correspond to the sizes of the operands; the `sgn` indicates the sign of the result.
+    pub fn split_mul<const RHS_LIMBS: usize>(
+        &self,
+        rhs: &Int<RHS_LIMBS>,
+    ) -> (Uint<{ LIMBS }>, Uint<{ RHS_LIMBS }>, Choice) {
```

Comment:
it is actually a way to make is slightly easier.
Here's as an example my implementation of widening_mul
```rust
    pub const fn widening_mul<const RHS_LIMBS: usize, const WIDE_LIMBS: usize>(
        &self,
        rhs: &Int<RHS_LIMBS>,
    ) -> Int<WIDE_LIMBS>
    where
        Uint<LIMBS>: ConcatMixed<Uint<RHS_LIMBS>, MixedOutput=Uint<WIDE_LIMBS>>,
    {
        let l_sign = self.is_negative();
        let l_pos = Uint::select(&self.repr, &self.repr.wrapping_neg(), l_sign);
        let r_sign = self.is_negative();
        let r_pos = Uint::select(&rhs.repr, &rhs.repr.wrapping_neg(), r_sign);
        let p = l_pos.widening_mul(&r_pos);
        
        Int { repr: Uint::select(&p, &p.wrapping_neg(), l_sign.xor(r_sign)) }
    }
}
```